pkg/platform/kube/functionres/lazy.go

```diff
@@ -1209,6 +1312,98 @@ func (lc *lazyClient) ensureServicePortsExist(to []v1.ServicePort, from []v1.Ser
 	return to
 }
 
+func (lc *lazyClient) populateCronJobConfig(functionLabels labels.Set,
+	function *nuclioio.NuclioFunction,
+	resources Resources,
+	cronTrigger functionconfig.Trigger,
+	meta *meta_v1.ObjectMeta,
+	spec *v1beta1.CronJobSpec) error {
+	var err error
+
+	lc.logger.DebugWith("Preparing cron job")
+
+	type cronAttributes struct {
+		Schedule          string
+		Interval          string
+		ConcurrencyPolicy string
+		Event             cron.Event
+	}
+
+	// get the attributes from the cron trigger
+	var attributes cronAttributes
+	if err = mapstructure.Decode(cronTrigger.Attributes, &attributes); err != nil {
+		return errors.Wrap(err, "Failed to decode cron trigger attributes")
+	}
+
+	// populate schedule
+	if attributes.Interval != "" {
+		spec.Schedule = fmt.Sprintf("@every %s", attributes.Interval)
+	} else {
+		spec.Schedule, err = common.NormalizeCronScheduleToFive(attributes.Schedule)
+		if err != nil {
+			return errors.Wrap(err, "Failed to normalize cron schedule")
+		}
+	}
+
+	// generate a string to be sent as the request body argument to curl
+	eventBodyAsCurlArg := ""
+	if attributes.Event.Body != "" {
+		eventBodyAsCurlArg = fmt.Sprintf("--post-data \"%s\"", attributes.Event.Body)
+	}
+
+	// generate a string containing all of the headers with -H flag as prefix, to be used by cURL later
+	headersAsCurlArg := ""
+	for headerKey, headerValue := range attributes.Event.Headers {
+		headerValueAsString, ok := headerValue.(string)
+		if !ok {
+			return errors.New(fmt.Sprintf("Unexpected header value type (expected string). header key: %s", headerKey))
+		}
+
+		headersAsCurlArg = fmt.Sprintf("%s --header \"%s: %s\"", headersAsCurlArg, headerKey, headerValueAsString)
+	}
+
+	// add default header
+	headersAsCurlArg = fmt.Sprintf("%s --header \"%s: %s\"", headersAsCurlArg, "x-nuclio-invoke-trigger", "cron")
+
+	// get the function http trigger address from the service
+	functionService, err := resources.Service()
+	if err != nil {
+		return errors.Wrap(err, "Failed to get function service")
+	}
+
+	functionAddress := fmt.Sprintf("%s:%s", functionService.Spec.ClusterIP, "8080")
+
+	// generate the whole curl command to be run by the CronJob to invoke the function
+	curlCommand := fmt.Sprintf("wget %s %s %s", eventBodyAsCurlArg, headersAsCurlArg, functionAddress)
+
+	spec.JobTemplate = v1beta1.JobTemplateSpec{
+		Spec: batchv1.JobSpec{
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: meta_v1.ObjectMeta{
+					Labels: map[string]string{"nuclio.io/function-cron-job-pod": "true"},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "function-invocator",
+							Image: "busybox",
```

Comment:
This needs to be controlled from helm chart like all other images + tags (offline usage)